src/main/java/com/icthh/xm/uaa/domain/converter/RoleKeyConverter.java

```diff
@@ -0,0 +1,75 @@
+package com.icthh.xm.uaa.domain.converter;
+
+
+import static com.google.common.collect.Iterables.getFirst;
+import static java.util.Collections.emptyList;
+import static org.apache.commons.lang3.StringUtils.isNoneBlank;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import com.icthh.xm.uaa.service.TenantPropertiesService;
+import java.io.IOException;
+import java.util.List;
+import javax.persistence.AttributeConverter;
+import javax.persistence.Converter;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Converter
+@Component
+@Slf4j
+public class RoleKeyConverter implements AttributeConverter<List<String>, String> {
+
+    private TenantPropertiesService tenantPropertiesService;
+    private ObjectMapper mapper = new ObjectMapper();
+
+    public RoleKeyConverter(){
+        mapper.registerModule(new JavaTimeModule());
+    }
+
+    @Override
+    public String convertToDatabaseColumn(List<String> s) {
+        return toJson(s);
+    }
+
+    @Override
+    public List<String> convertToEntityAttribute(String roles) {
+        List<String> multiRoles = fromJson(roles);
+        if(multiRoles.isEmpty()) {
+            return multiRoles;
+        }
+        return tenantPropertiesService.getTenantProps().isMultiRoleEnabled() ?
+            multiRoles :
+            List.of(getFirst(multiRoles, null));
```

Comment:
Is it OK if List.of(null) will fail with NPE?

Reply:
fixed